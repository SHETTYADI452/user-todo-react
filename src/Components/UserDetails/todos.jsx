// importing modules

import React from "react";
import ClipLoader from "react-spinners/ClipLoader";
import "../../styles/userdetails.css";

//exporting modules todos and isloding which is being used in  userdetails index files
export const Todos = ({ todos, isLoading }) => {
  return (
    <div>
      {isLoading ? (
        <div>
          <h3 className="loader">Fetching Todos...</h3>
          <ClipLoader loading={isLoading} size={150} />
        </div>
      ) : (
        <table id="todo-table">
          <thead>
            <tr>
              <th>Todo Id</th>
              <th>Todo Task Title</th>
              <th>Completed</th>
            </tr>
          </thead>
          <tbody>
            {/* Mapping the todo data  */}
            {todos.map((todo) => {
              return (
                <tr key={todo.id}>
                  <td>{todo.id}</td>
                  <td id={`title-${todo.id}`}>{todo.title}</td>
                  <td>{todo.completed ? "Completed" : "Incomplete"}</td>
                </tr>
              );
            })}
          </tbody>
        </table>
      )}
    </div>
  );
};
