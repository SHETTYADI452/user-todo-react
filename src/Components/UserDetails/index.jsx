// Importing modules
import React, { useContext, useEffect, useState } from "react";
import { UsersContext } from "../../Contexts/Users";
import { LoadingContext } from "../../Contexts/Loading";
import { fetchTodo, addNewTodo } from "../../actions/todo";
import { Todos } from "./todos";
import { setSelectedUser } from "../../actions/users";
import { useParams } from "react-router-dom";
import { TodoContext } from "../../Contexts/Todo";
import { TextField } from "../Common/TextField";
import "../../styles/userdetails.css";

//exporting  modules
export const UserDetails = () => {
  const title = "User Details Page";

  // initializing user context
  const { state: userState, dispatch: userDispatch } = useContext(UsersContext);
  // initializing loading context
  const { state: loadingState, dispatch: loadingDispatch } =
    useContext(LoadingContext);

  // initializing Todo context
  const { state: todoState, dispatch: todoDispatch } = useContext(TodoContext);
  const { id } = useParams();
  const [newTodo, setNewTodo] = useState();

  useEffect(() => {
    setSelectedUser(id)(userDispatch, loadingDispatch);
    fetchTodo(id)(todoDispatch, loadingDispatch);
  }, []);

  // this function is used to add new todo task after entering an input and performing click event
  const handleClick = () => {
    if (!newTodo) return;
    const userData = {
      id,
      title: newTodo,
      completed: false,
    };
    addNewTodo(userData)(todoDispatch, loadingDispatch);
    setNewTodo("");
  };

  return (
    <div>
      <center>
        <h1 id="page-title">{title}</h1>
      </center>
      <div className="usercontainer">
        <div className="userdetail">
          <div className="ud">
            <h2>
              <b>Name : </b>
              {userState?.user?.name ?? ""}
            </h2>
            <div>
              <b>User Name :</b> {userState?.user?.username ?? ""}
            </div>
            <div>
              <b>Email Id :</b> {userState?.user?.email ?? ""}
            </div>
            <div>
              <b>Address :</b> {userState?.user?.address?.suite ?? ""},
              {userState?.user?.address?.street ?? ""},
              {userState?.user?.address?.city ?? ""}
            </div>
            <div>
              <b>Company Name :</b> {userState?.user?.company?.name ?? ""}
            </div>
          </div>
          <div className="add-todo">
            <TextField
              id={"todo-input"}
              label={"Add Todo"}
              placeholder={"Add Task"}
              value={newTodo}
              // the setTodoInput will set the changes occur in input field
              handleChange={(e) => setNewTodo(e.target.value)}
            />
            <button id="add-btn" onClick={handleClick}>
              Add
            </button>
          </div>
        </div>

        <div className="usertodo">
          <center>
            <h2>Users Todo's</h2>
            <Todos todos={todoState.todos} isLoading={loadingState.isLoading} />
          </center>
        </div>
      </div>
    </div>
  );
};
