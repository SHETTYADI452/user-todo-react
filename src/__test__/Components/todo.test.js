import { mount } from "enzyme";
import { Todos } from "../../Components/UserDetails/todos";

describe("Testing TodoList component", () => {
  let wrapper;
  let secondaryWrapper;
  const todo = [
    {
      userId: 1,
      id: 1,
      title: "Test",
      completed: true,
    },
    {
      userId: 1,
      id: 2,
      title: "Test2",
      completed: false,
    },
  ];

  beforeEach(() => {
    wrapper = mount(<Todos todos={todo} />);
  });
  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should check for loader existence", () => {
    secondaryWrapper = mount(<Todos todos={todo} isLoading={true} />);
    expect(secondaryWrapper).toMatchSnapshot();
  });

  it("should have the todo in the list", () => {
    expect(wrapper.find(`#title-${todo[0].id}`).text()).toBe(todo[0].title);
  });

  it("should have a page loader", () => {
    const pageLoader = wrapper.find(".loader");
    expect(pageLoader).toBeTruthy;
  });
});
