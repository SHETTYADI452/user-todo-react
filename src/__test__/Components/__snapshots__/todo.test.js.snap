// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TodoList component should check for loader existence 1`] = `ReactWrapper {}`;

exports[`Testing TodoList component should match the snapshot 1`] = `ReactWrapper {}`;
